app/controllers/home_controller.rb

```diff
@@ -7,8 +7,8 @@ def index
     @matches = []
 
     all_matches_for_current_user.each do |match|
-      username = match.user_a_id != current_user.id ? match.user_a.name : match.user_b.name
-      @matches.push({ id: match.id, username: username, topic: match.topic.name })
+      user = match.user_a_id != current_user.id ? match.user_a : match.user_b
```

Comment:
seems like duplicated code: that method could be extracted to application_controller, or the model